apps/wallet-mobile/src/features/Portfolio/useCases/PortfolioTokenDetails/PortfolioTokenDetailsScreen.tsx

```diff
@@ -72,37 +71,38 @@ export const PortfolioTokenDetailsScreen = () => {
   }, [activeTab, setActiveTab, strings.overview, strings.performance, strings.transactions, styles.tab, styles.tabs])
 
   return (
-    <PortfolioTokenDetailLayout
-      topContent={
-        <Animated.View style={[styles.tabsSticky, isStickyTab ? styles.tabsStickyActive : styles.tabsStickyInactive]}>
-          {renderTabs}
-        </Animated.View>
-      }
-      onScroll={onScroll}
-      ListHeaderComponent={
-        <>
-          <Animated.View style={styles.header}>
-            <Spacer height={16} />
-
-            <PortfolioTokenBalance />
-
-            <Spacer height={16} />
-
-            <PortfolioTokenChart />
-
-            <Spacer height={16} />
-          </Animated.View>
-
-          <Animated.View>{renderTabs}</Animated.View>
-
-          <PortfolioTokenInfo />
-        </>
-      }
-      ListFooterComponent={loadingView}
-      {...getSectionListProps}
-    >
+    <TxFilter tokenId={tokenId}>
+      <Animated.View style={[styles.tabsSticky, isStickyTab ? styles.tabsStickyActive : styles.tabsStickyInactive]}>
+        {renderTabs}
+      </Animated.View>
+
+      <TxList
+        onScroll={onScroll}
+        ListEmptyComponent={<BuyADABanner />}
+        ListHeaderComponent={
+          <>
+            <Animated.View style={styles.header}>
+              <Spacer height={16} />
+
+              <PortfolioTokenBalance />
+
+              <Spacer height={16} />
+
+              <PortfolioTokenChart />
+
+              <Spacer height={16} />
+            </Animated.View>
+
+            <Animated.View>{renderTabs}</Animated.View>
+
+            <PortfolioTokenInfo />
+          </>
+        }
+        {...(activeTab !== 'transactions' && {data: []})}
```

Comment:
I don't think there's a better way to set data to empty, and not send the data prop otherwise. See previous comment as to why the whole screen can't be null.